### PR TITLE
[Feature] Add ResizeShortestEdge transform

### PR DESCRIPTION
**Pull request by xiexinch**
## Motivation

Support resize shortest edge while keeping the aspect ratio unchanged.

Modifed from https://github.com/facebookresearch/detectron2/blob/main/detectron2/data/transforms/augmentation_impl.py#L130

## Modification

- Add `ResizeShortestEdge` to mmseg/datasets/transforms/transforms.py
- Add unit test

## BC-breaking (Optional)

Depend on `mmcv >= 2.0.0rc3`

## Use cases (Optional)

At `train_pipeline`:

```python
train_pipeline = [
    ...
    dict(
        type='RandomChoiceResize',
        scales=[128, 256, 512],
        resize_type='ResizeShortestEdge',
        max_size=2048,
    )
    ...
]
```

### CODE REVIEW

File: mmseg/datasets/transforms/transforms.py [hunk 1]
```diff
@@ -1226,3 +1226,75 @@ def __repr__(self):
...
+
+        # Create a empty Resize object
+        self.resize = TRANSFORMS.build({
+            'type': 'Resize',
+            'scale': 0,
+            'keep_ratio': True
+        })
+        self.scale = scale
+
+    def _get_output_shape(self, img, short_edge_length) -> Tuple[int, int]:
```
Comment (MeowZheng) [inline mmseg/datasets/transforms/transforms.py:1277]: please add docstring

File: mmseg/datasets/transforms/transforms.py [hunk 2]
```diff
@@ -1226,3 +1226,75 @@ def __repr__(self):
...
+        h, w = img.shape[:2]
+        if isinstance(short_edge_length, int):
+            size = short_edge_length * 1.0
+        elif isinstance(short_edge_length, tuple):
+            size = min(short_edge_length) * 1.0
```
Comment (MeowZheng) [inline mmseg/datasets/transforms/transforms.py:1282]: `short_edge_length` can be an `int` or a tuple of `int`?
if `short_edge_length` can be a tuple, please add some description in `__init__` docstring for `scale`

File: tests/test_datasets/test_transform.py [hunk 3]
```diff
@@ -71,6 +71,37 @@ def test_resize():
     resized_results = resize_module(results.copy())
     assert max(resized_results['img_shape'][:2]) <= 1333 * 1.1
 
+    # test RandomChoiceResize, which `resize_type` is `ResizeShortestEdge`
+    transform = dict(
+        _scope_='mmseg',
```
Comment (MeowZheng) [inline tests/test_datasets/test_transform.py:76]: why add `_scope_='mmseg'` here?

File: mmseg/datasets/transforms/transforms.py [hunk 4]
```diff
@@ -1226,3 +1226,75 @@ def __repr__(self):
         repr_str += f'edge_width={self.edge_width}, '
         repr_str += f'ignore_index={self.ignore_index})'
         return repr_str
+
+
+@TRANSFORMS.register_module()
+class ResizeShortestEdge(BaseTransform):
+    """Resize the image and mask while keeping the aspect ratio unchanged.
+
+    Modified from https://github.com/facebookresearch/detectron2/blob/main/detectron2/data/transforms/augmentation_impl.py#L130  # noqa:E501
```
Comment (MeowZheng) [inline mmseg/datasets/transforms/transforms.py:1235]: please add copyright and license

File: mmseg/datasets/transforms/transforms.py (hunk 2 above)
Comment (xiexinch) [reply mmseg/datasets/transforms/transforms.py:1282]: It's recommended to be `int`, but the entry `RandomChoiceResize` accept both `int` and `tuple` as input.

https://github.com/open-mmlab/mmcv/blob/2.x/mmcv/transforms/processing.py#L1057

File: mmseg/datasets/transforms/transforms.py [hunk 5]
```diff
@@ -1226,3 +1226,85 @@ def __repr__(self):
...
+            new_h, new_w = scale * h, size
+
+        if max(new_h, new_w) > self.max_size:
+            scale = self.max_size * 1.0 / max(new_h, new_w)
+            new_h *= scale
+            new_w *= scale
+
+        new_h = int(new_h + 0.5)
+        new_w = int(new_w + 0.5)
+        return (new_h, new_w)
```
Comment (MeowZheng) [inline mmseg/datasets/transforms/transforms.py:1306]: It seems like the input `scale` of `Resize` in MMCV is `(w, h)`

**Review by MeowZheng — APPROVED** [review]

**Pull request merged**